lib/merkle-tree.js

```diff
@@ -1224,7 +1227,25 @@ function getStoredNode (storage, index, cache, error) {
       }
 
       const node = { index, size, hash }
-      if (cache !== null) cache.set(index, node)
+
+      if (cache !== null) {
+        const slab = b4a.alloc(nodeSize)
+        b4a.copy(data, slab)
+
+        const sizeCopy = c.decode(
```

Comment:
I'm not sure if it's needed to copy the size. When decoding a buffer into an integer, is the original buffer still referenced? (if not, only the hash needs to be copied)

Reply:
we can just do this:

```js
const node = .... // like before

if (cache !== null) {
  node.hash = b4a.from(node.hash) // copy
  cache.set(index, node)
}
```

as the hash is the only thing keeping the slab alive

Reply:
Nice, not having to copy the int makes it easier.

`b4a.from(...)` does seem to use slab memory though, the test shows its underlying buffer has length ~8kb, so I just explicitly copied the hash (and nothing else)